java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -1874,6 +1876,31 @@ protected int diffSwitch(JCSwitch oldT, JCSwitch newT, int[] bounds) {
         return bounds[1];
     }
 
+    protected int diffSwitchExpression(Tree oldT, Tree newT, int[] bounds) {
```

Comment:
I think it would be nice to have tests specifically for the code generator testing things work w.r.t. switch expressions. I get this is tested indirectly through the hints, but:
-hints tests often ignore whitespace
-when modifying the codegen and checking if the change is OK, it is easier to run the codegen tests than to run the java.hints.
-the codegen tests can also test things/changes that java.hints don't use.

Please see ```java.source.base/test/unit/src/org/netbeans/api/java/source/gen```

Reply:
@jlahoda Thanks for providing the valuable feedback. I will check and make the corresponding changes.

Reply:
Test cases added